src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureVirtualMachineOperationSDK.java

```diff
@@ -0,0 +1,73 @@
+package cloud.fogbow.ras.core.plugins.interoperability.azure.compute.sdk;
+
+import cloud.fogbow.common.exceptions.InstanceNotFoundException;
+import cloud.fogbow.common.exceptions.NoAvailableResourcesException;
+import cloud.fogbow.common.exceptions.UnauthenticatedUserException;
+import cloud.fogbow.common.exceptions.UnexpectedException;
+import cloud.fogbow.common.models.AzureUser;
+import cloud.fogbow.ras.constants.Messages;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.compute.AzureGetVirtualMachineRef;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.compute.AzureVirtualMachineOperation;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.util.AzureClientCacheManager;
+import com.google.common.annotations.VisibleForTesting;
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.compute.VirtualMachine;
+import com.microsoft.azure.management.compute.VirtualMachineSize;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import org.apache.log4j.Logger;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class AzureVirtualMachineOperationSDK implements AzureVirtualMachineOperation {
+
+    private static final Logger LOGGER = Logger.getLogger(AzureVirtualMachineOperationSDK.class);
+
+    @Override
+    public AzureGetVirtualMachineRef doGetInstance(String azureInstanceId, AzureUser azureCloudUser)
+            throws UnauthenticatedUserException, UnexpectedException,
+            NoAvailableResourcesException, InstanceNotFoundException {
+
+        Azure azure = AzureClientCacheManager.getAzure(azureCloudUser);
+
+        VirtualMachine virtualMachine = AzureVirtualMachineSDK
+                .getVirtualMachineById(azure, azureInstanceId)
+                .orElseThrow(InstanceNotFoundException::new);
```

Comment:
It would be interesting to follow a pattern in using elseThrow(). In this line it is passed as parameter `InstanceNotFoundException::new`, but in line 70, it is used as `() -> new NoAvailableResourcesException()`.

Reply:
Done. 
I've made the change in the branch azure-compute-plugin-four-pr-fixes. [Commit reference](https://github.com/fogbow/resource-allocation-service/commit/7999b929d1c5db271e566b43aaa4a88f2bc48187).